R/rescale_weights.R

```diff
@@ -2,96 +2,159 @@
+#' data(nhanes_sample)
+#' head(rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR"))
+#'
+#' # also works with multiple group-variables
+#' head(rescale_weights(nhanes_sample, c("SDMVSTRA", "SDMVPSU"), "WTINT2YR"))
+#'
+#' # or nested structures.
+#' x <- rescale_weights(
+#'   data = nhanes_sample,
+#'   by = c("SDMVSTRA", "SDMVPSU"),
+#'   probability_weights = "WTINT2YR",
+#'   nest = TRUE
+#' )
+#' head(x)
+#'
+#' \donttest{
+#' # compare different methods, using multilevel-Poisson regression
 #'
-#' @examples
-#' if (require("lme4")) {
-#'   data(nhanes_sample)
-#'   head(rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR"))
-#'
-#'   # also works with multiple group-variables
-#'   head(rescale_weights(nhanes_sample, c("SDMVSTRA", "SDMVPSU"), "WTINT2YR"))
-#'
-#'   # or nested structures.
-#'   x <- rescale_weights(
-#'     data = nhanes_sample,
-#'     by = c("SDMVSTRA", "SDMVPSU"),
-#'     probability_weights = "WTINT2YR",
-#'     nest = TRUE
-#'   )
-#'   head(x)
-#'
-#'   nhanes_sample <- rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR")
-#'
-#'   glmer(
-#'     total ~ factor(RIAGENDR) * (log(age) + factor(RIDRETH1)) + (1 | SDMVPSU),
-#'     family = poisson(),
-#'     data = nhanes_sample,
-#'     weights = pweights_a
-#'   )
+#' d <- rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR")
+#' result1 <- lme4::glmer(
+#'   total ~ factor(RIAGENDR) + log(age) + factor(RIDRETH1) + (1 | SDMVPSU),
+#'   family = poisson(),
+#'   data = d,
+#'   weights = rescaled_weights_a
+#' )
+#' result2 <- lme4::glmer(
+#'   total ~ factor(RIAGENDR) + log(age) + factor(RIDRETH1) + (1 | SDMVPSU),
+#'   family = poisson(),
+#'   data = d,
+#'   weights = rescaled_weights_b
+#' )
+#'
+#' d <- rescale_weights(
+#'   nhanes_sample,
+#'   probability_weights = "WTINT2YR",
+#'   method = "kish"
+#' )
+#' result3 <- lme4::glmer(
+#'   total ~ factor(RIAGENDR) + log(age) + factor(RIDRETH1) + (1 | SDMVPSU),
+#'   family = poisson(),
+#'   data = d,
+#'   weights = rescaled_weights
+#' )
+#' parameters::compare_parameters(
+#'   list(result1, result2, result3),
+#'   exponentiate = TRUE,
+#'   column_names = c("Carle (A)", "Carle (B)", "Kish")
+#' )
 #' }
 #' @export
-rescale_weights <- function(data, by, probability_weights, nest = FALSE) {
+rescale_weights <- function(data,
+                            by = NULL,
+                            probability_weights = NULL,
+                            nest = FALSE,
+                            method = "carle") {
   if (inherits(by, "formula")) {
     by <- all.vars(by)
   }
 
+  # check for existing variable names
+  if ((method == "carle" && any(c("rescaled_weights_a", "rescaled_weights_b") %in% colnames(data))) ||
+    (method == "kish" && "rescaled_weights" %in% colnames(data))) {
+    insight::format_warning("The variable name for the rescaled weights already exists in the data. Returned columns will be renamed into unique names.") # nolint
```

Comment:
Should this be an error? No particular pref, but it seems like it could be a footgun

Reply:
I have no particular preference either. Since we "decide" on the name of the new column, I thought it would be fair to just warn, and give the rescaled weights column a different name than the default one, if it already exists in the data